spec/lib/vanagon/project_spec.rb

```diff
@@ -126,6 +127,74 @@
     end
   end
 
+  describe "#load_yaml_settings" do
+    subject(:project) do
+      project = Vanagon::Project.new('yaml-inheritance-test', Vanagon::Platform.new('aix-7.2-ppc'))
+      project.settings = { merged: 'nope', original: 'original' }
+      project
+    end
+
+    let(:yaml_filename) { 'settings.yaml' }
+    let(:sha1_filename) { "#{yaml_filename}.sha1" }
+
+    let(:yaml_path) { "/path/to/#{yaml_filename}" }
+    let(:sha1_path) { "/path/to/#{sha1_filename}" }
+
+    let(:yaml_content) { { other: 'other', merged: 'yup' }.to_yaml }
+    let(:sha1_content) { "abcdef output/tarball.tar.gz" }
+
+    let(:git_uri) { 'git://some.url' }
+    let(:local_yaml_uri) { "file://#{yaml_path}" }
+    let(:http_yaml_uri) { "http:/#{yaml_path}" }
+    let(:http_sha1_uri) { "http:/#{sha1_path}" }
+
+    before(:each) do
+      allow(Dir).to receive(:mktmpdir) { |&block| block.yield '' }
+    end
+
+    it "fails for a local source if the settings file doesn't exist" do
+      expect { project.load_yaml_settings(local_yaml_uri) }.to raise_error(Vanagon::Error)
+    end
+
+    it "fails if given a git source" do
```

Comment:
@caseywilliams Yeah, this is pretty much what I was thinking. My only other suggestion is maybe adding another test to ensure it fails for an `:unknown` source since the routine also includes a check for that.

Reply:
@ekinanp updated, and squashed the tests updates into the original commit